qmi/core/transport.py

```diff
@@ -179,16 +182,30 @@ def discard_read(self) -> None:
 
 
 class TransportDescriptorParser:
-
+    """This class is for creating a transport-specific parser classes and has (static) methods that are
+    used for parsing transport strings.
+    """
     def __init__(self,
                  interface: str,
                  positionals: List[Tuple[str, Tuple[Type, bool]]],
-                 keywords: Mapping[str, Tuple[Type, bool]]):
+                 keywords: Mapping[str, Tuple[Type, bool]]
+    ) -> None:
         self.interface = interface
         self._positionals = positionals
         self._keywords = keywords
 
-    def parse_parameter_strings(self, transport_descriptor: str, default_parameters=None) -> Mapping[str, Any]:
+    def parse_parameter_strings(
+            self, transport_descriptor: str, default_parameters: Optional[Dict[str, Any]] = None
+    ) -> Mapping[str, Any]:
```

Comment:
the return type here would also be a `Dict`

Reply:
If Mypy does not complain, it should be fine like this. I'll leave it for now as 'original' as I get no trouble with it.

Reply:
hmmm, with this typing we are implying 2 things:
1. somewhere in this function we have changed the `parameters` from `Dict` to `Mapping` (as we have changed the typing for the other private methods from `Mapping` to `Dict`)
2. `Mapping` is immutable and `Dict` is not, so the parameters we return are immutable according to our function signature

Reply:
I see, that is misleading indeed. It can be changed to `MutableMapping` to solve this issue.

Reply:
I think that will work. `Dict` inherits from `MutableMapping` (which is an ABC), but for typing I don't think that should make a difference. But to be safe and correct I still prefer `Dict`, because inside this method `parameters` is always of type `Dict`

Reply:
OK, I'll change it.